### DISCUSSION TITLE
send file from client to server

### DISCUSSION THREAD

**Discussion by tobias-carlbom** [jchristn/WatsonTcp#225] [Q&A] [answered]
howdy,

i have spent several days working with watsontcp and it has been a pleasure! you have done an awesome job!!

i would like to send a file from the tcp client to the tcp server (different computers). i guess that i need to use the watson stream functionallity but i am not quit sure how.

have someone here tried to implement this or can give me some tips? :)

**Comment by jchristn** [comment, marked as answer]
Hi @manprinsen I just pushed a commit that includes a project called ```Test.FileTransfer```, perhaps it will be useful for you.  Cheers

**Reply by tobias-carlbom** [reply]
@jchristn wow thats so awesome!! thank you so muck 😀 🙏
just one more question, it seems that it is not possible to both use e.g. server.Events.MessageReceived += MessageReceived; and server.Events.StreamReceived += ServerStreamReceived;

it seems that the MessageReceived callback is always triggered when client sends message to server.

**Reply by jchristn** [reply]
Hi, yes that's right. You can only use one or the other.

**Reply by tobias-carlbom** [reply]
Alright :) I’ll play around a bit more hoping to find a way that suits me :)

Thanks again 🙏

**Reply by tobias-carlbom** [reply]
sorry for spamming you @jchristn !
would it be possible to send files with version 4.8.14.14? im willing to pay for an example :)

it seems that _Client.Send(len,fs,md) is only for > version 5 :/
https://github.com/jchristn/WatsonTcp/blob/master/src/Test.FileTransfer/Program.cs (looking at "client send file")

**Reply by jchristn** [reply]
That API also exists in v4.x.  The Test.FileTransfer project was not added to the solution (and to github) until 5.x.
